### ISSUE TITLE
SpecCluster force scheduler to None

### ISSUE THREAD

**Issue by davzucky** [dask/distributed#5650]
**What happened**:
When you pass in the ```SpecCluster``` a ```scheduler``` already initialized you expect the cluster to use this instance. However the code force it to ```None``` and only extract the spec of the cluster. https://github.com/dask/distributed/blob/1bab9e54b2199f0b49934b7e84e10b449cd5fef2/distributed/deploy/spec.py#L234

**What you expected to happen**:
I expect the ```SpecCluster``` to initialize the ```scheduler``` property with the one pass as input. Later on the ```SpecCluster._start``` the code check if the ```scheduler``` is already initialize which will never be the case https://github.com/dask/distributed/blob/1bab9e54b2199f0b49934b7e84e10b449cd5fef2/distributed/deploy/spec.py#L284

**Minimal Complete Verifiable Example**:
I'm trying the test the new plugin from that issue on dask-kubernetes https://github.com/dask/dask-kubernetes/issues/377 as per @jacobtomlinson recommendation. However this issue above prohibit me from reusing current logic to do it.

The code below is the new implementation I'm trying to get working. This is a WIP 

```python
from distributed.diagnostics import SchedulerPlugin
from dask_kubernetes.core import KubeCluster
from typing import Union
from  kubernetes.client.models.v1_pod import V1Pod


class KubernetesAutoScaler(SchedulerPlugin):
    def __init__(self, pod_spec: Union[V1Pod, dict, str], min_workers: int, max_workers: int) -> None:
        self._min_workers = min_workers
        self._max_workers = max_workers
        self._pod_spec = pod_spec
        super().__init__()
    
    async def start(self, scheduler):
        self._cluster = KubeCluster(self._pod_spec, scheduler = scheduler)
        self._adaptive = self._cluster.adapt(minimum=self._min_workers, maximum=self._max_workers)
        return await super().start(scheduler)

    async def close(self):
        self._cluster.close(30)
        await super().close()
   ```

**Comment by jacobtomlinson** [comment]
The `SpecCluster` class expects a class, not an instance of a class. It will do the initialization itself. 

You'll either need to subclass `Scheduler` and make your customisations there or modify the `cluster.scheduler` attribute once you have initialized `SpecCluster`.

**Comment by jacobtomlinson** [comment]
I'm pretty confident that the operator work being done will solve this problem for you. So perhaps you would be interested in helping us test it as we are working on it.

**Comment by davzucky** [comment]
Yes for sure I will test it even if this is only on my local env. Really looking forward to seeing that operator. At the same time, I will continue to explore the plugin option which give me the opportunity to dive into Dask code
